### ISSUE TITLE
remove_reference_t is not defined when `__remove_reference` builtin is expressed.

### ISSUE THREAD

**Issue by seanbaxter** [llvm/llvm-project#53569]
https://github.com/llvm/llvm-project/blob/main/libcxx/include/type_traits#L1287

Forgot the alias template for the builtin case.

**Comment by ldionne** [comment]
Confirmed.

Actually, Clang doesn't implement that builtin so this is effectively dead code. Several other type traits suffer from the same problem.

**Comment by ldionne** [comment]
https://reviews.llvm.org/D118923

**Comment by ldionne** [comment]
```
commit abcbe87311db67cfee47478f2b97b4ed396df5e6
Author: Louis Dionne <ldionne.2@gmail.com>
Date:   Thu Feb 3 12:29:35 2022 -0500

    [libc++] Remove unused code paths for non-existent builtins

    It looks like we added some checks to try and use builtin type traits
    in https://reviews.llvm.org/D67900, but some of those type traits are
    never implemented as builtins, so this is essentially dead code.

    Fixes llvm-project#53569

    Differential Revision: https://reviews.llvm.org/D118923
```
